blocks/api/registration.js

```diff
@@ -43,6 +43,13 @@ export function registerBlock( slug, settings ) {
 		);
 		return;
 	}
+	if ( true !== validateBlockSettings( settings ) ) {
+		console.error(
+			'Block not registered.'
+		);
+		// Return the block even though it was not registered.
+		return Object.assign( { slug }, settings );
```

Comment:
Thinking since we already assign this as `block` below that we could move the condition after the variable assignment and simply return the variable instead of duplicating the logic (especially in case we change the merge logic down the road).

```js
const block = Object.assign( { slug }, settings );
if ( ! isValidBlockSettings( settings ) ) {
	console.error(
		'Block not registered.'
	);
	// Return the block even though it was not registered.
	return block;
}
```

Or perhaps we reflect the intent that the global `blocks` object should only assign the new block if it's valid, but allow the function to continue to the final `return`:

```js
const block = Object.assign( { slug }, settings );
if ( isValidBlockSettings( settings ) ) {
	blocks[ slug ] = block;
} else {
	console.error(
		'Block not registered.'
	);
}

return block;
```